m2cgen/assemblers/boosting.py

```diff
@@ -225,6 +237,76 @@ def _final_transform(self, ast_to_transform):
         else:
             return super()._final_transform(ast_to_transform)
 
+    def _multi_class_convert_output(self, exprs):
+        supported_objectives = {
+            "multiclass": super()._multi_class_convert_output,
+            "multiclassova": self._multi_class_sigmoid_transform,
+            "custom": super()._single_convert_output,
+        }
+        if self.objective_name not in supported_objectives:
+            raise ValueError(
+                f"Unsupported objective function '{self.objective_name}'")
+        return supported_objectives[self.objective_name](exprs)
+
+    def _multi_class_sigmoid_transform(self, exprs):
+        return [self._bin_class_sigmoid_transform(expr, to_reuse=False)
+                for expr in exprs]
+
+    def _bin_class_convert_output(self, expr, to_reuse=True):
+        supported_objectives = {
+            "binary": self._bin_class_sigmoid_transform,
+            "custom": super()._single_convert_output,
+        }
+        if self.objective_name not in supported_objectives:
+            raise ValueError(
+                f"Unsupported objective function '{self.objective_name}'")
+        return supported_objectives[self.objective_name](expr)
+
+    def _bin_class_sigmoid_transform(self, expr, to_reuse=True):
+        coef = 1.0
+        for config_part in self.objective_config.split(" "):
+            config_entry = config_part.split(":")
+            if config_entry[0] == "sigmoid":
+                coef = np.float64(config_entry[1])
+                break
+        return super()._bin_class_convert_output(
+            utils.mul(ast.NumVal(coef), expr) if coef != 1.0 else expr,
+            to_reuse=to_reuse)
+
+    def _single_convert_output(self, expr):
+        supported_objectives = {
+            "cross_entropy": fallback_expressions.sigmoid,
+            "cross_entropy_lambda": self._log1p_exp_transform,
+            "regression": self._maybe_sqr_transform,
+            "regression_l1": self._maybe_sqr_transform,
+            "huber": super()._single_convert_output,
+            "fair": self._maybe_sqr_transform,
+            "poisson": self._exp_transform,
+            "quantile": self._maybe_sqr_transform,
+            "mape": self._maybe_sqr_transform,
+            "gamma": self._exp_transform,
+            "tweedie": self._exp_transform,
+            "custom": super()._single_convert_output,
+        }
+        if self.objective_name not in supported_objectives:
+            raise ValueError(
+                f"Unsupported objective function '{self.objective_name}'")
+        return supported_objectives[self.objective_name](expr)
+
+    def _log1p_exp_transform(self, expr):
+        return ast.Log1pExpr(ast.ExpExpr(expr))
+
+    def _maybe_sqr_transform(self, expr):
+        need_sqr = "sqrt" in self.objective_config.split(" ")
+        if need_sqr:
+            expr = ast.IdExpr(expr, to_reuse=True)
+            return utils.mul(ast.AbsExpr(expr), expr)
```

Comment:
I don't quite understand the purpose of the Abs here. Can you please elaborate this expression

Reply:
Sure! We should preserve a sign of original result according to the original code. I decided to go with `abs` expression (instead of `sign`) because I think it is more common and may be used in future assemblers.

> `output[0] = Common::Sign(input[0]) * input[0] * input[0];`

https://github.com/microsoft/LightGBM/blob/4f8c32d9a6c6b0c8d774d571da3bcd70191d218b/src/objective/regression_objective.hpp#L146-L152

Reply:
Makes sense, thank you!